src/Microsoft.AspNet.OData.Shared/Builder/ParameterConfiguration.cs

```diff
@@ -51,5 +51,44 @@ protected ParameterConfiguration(string name, IEdmTypeConfiguration parameterTyp
         /// Gets or sets a value indicating whether this parameter is nullable or not.
         /// </summary>
         public bool Nullable { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this parameter is optional or not.
+        /// </summary>
+        public bool Optional { get; set; }
+
+        /// <summary>
+        /// Gets or sets a default value for optional parameter.
+        /// </summary>
+        public string DefaultValue { get; set; }
+
+        /// <summary>
+        /// Sets the optional value as true.
+        /// </summary>
+        public ParameterConfiguration IsOptional()
+        {
+            Optional = true;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the optional value as false.
+        /// </summary>
+        public ParameterConfiguration IsRequired()
```

Comment:
Same comment for IsRequired naming: I think something like SetRequired would be more clear.